lib/improver/tests/spotdata/spotdata/test_main.py

```diff
@@ -0,0 +1,196 @@
+                                         (latitude, 1),
+                                         (longitude, 2)],
+                    units="K")
+
+        orography = Cube(np.ones((20, 20)),
+                         long_name="surface_altitude",
+                         dim_coords_and_dims=[(latitude, 0),
+                                              (longitude, 1)],
+                         units="m")
+
+        self.ancillary_data = {}
+        self.ancillary_data.update({'orography': orography})
+
+        diagnostic_recipe = {
+            "temperature": {
+                "diagnostic_name": "air_temperature",
+                "extrema": False,
+                "filepath": "temperature_at_screen_level",
+                "interpolation_method": "use_nearest",
+                "neighbour_finding": {
+                    "land_constraint": False,
+                    "method": "fast_nearest_neighbour",
+                    "vertical_bias": None
+                    }
+                }
+            }
+
+        self.data_directory = mkdtemp()
+        self.output_directory = mkdtemp()
+
+        self.data_path = (self.data_directory +
+                          '/temperature_at_screen_level.nc')
+        self.orography_path = self.data_directory + '/orography.nc'
+        iris.save(cube, self.data_path)
+        iris.save(orography, self.orography_path)
+
+        self.config_path = self.data_directory + '/spotdata_constants.json'
+        ff = open(self.config_path, 'w')
+        json.dump(diagnostic_recipe, ff, sort_keys=True, indent=4,
+                  separators=(',', ': ',))
+        ff.close()
+
+        self.cube = cube
+        self.diagnostic_recipe = diagnostic_recipe
+
+        self.sites = OrderedDict()
+        self.sites['100'] = {'latitude': 50,
+                             'longitude': 0,
+                             'altitude': 10,
+                             'utc_offset': 0
+                             }
+
+        site_properties = [self.sites['100']]
+
+        self.args = (self.config_path, self.data_directory,
+                     self.data_directory)
+
+        self.kwargs = {
+            'diagnostic_list': ['temperature'],
+            'site_path': None,
+            'constants_file_path': None,
+            'site_properties': site_properties,
+            'forecast_date': '20170217',
+            'forecast_time': 6,
+            'forecast_length': 2,
+            'output_path': self.output_directory,
+            'use_multiprocessing': False
+            }
+
+    def tearDown(self):
+        """Remove temporary directories created for testing."""
+        Call(['rm', '-f', self.data_path])
+        Call(['rm', '-f', self.orography_path])
+        Call(['rm', '-f', self.config_path])
+        Call(['rm', '-f', self.output_directory + '/air_temperature.nc'])
+
+        Call(['rmdir', self.data_directory])
+        Call(['rmdir', self.output_directory])
+
+
+class Test_run_spotdata(Test_main):
```

Comment:
Can you add a test to show that this has been successful? I can see that this is a bit difficult because nothing is returned, as the file is written down in `WriteOutput`. From a testing perspective, it's a shame that the cubes aren't passed back up and the output isn't written out at a higher level, ideally in a spot data CLI.